core/variant/variant_utility.cpp

```diff
@@ -223,7 +223,40 @@ struct VariantUtilityFunctions {
 		return Math::snapped(value, step);
 	}
 
-	static inline double lerp(double from, double to, double weight) {
+	static inline Variant lerp(const Variant &from, const Variant &to, double weight, Callable::CallError &r_error) {
+		r_error.error = Callable::CallError::CALL_OK;
+		if (from.get_type() != to.get_type()) {
+			r_error.error = Callable::CallError::CALL_ERROR_INVALID_ARGUMENT;
+			return Variant();
+		}
```

Comment:
This should handle errors in case no automatic conversion should be allowed:
```suggestion
		if (from.get_type() != Variant::FLOAT && from.get_type() != Variant::VECTOR2 && from.get_type() != Variant::VECTOR3 && from.get_type() != Variant::QUATERNION && from.get_type() != Variant::BASIS && from.get_type() != Variant::COLOR) {
			r_error.error = Callable::CallError::CALL_ERROR_INVALID_ARGUMENT;
			r_error.argument = 0;
			return Variant();
		}
		if (from.get_type() != to.get_type()) {
			r_error.error = Callable::CallError::CALL_ERROR_INVALID_ARGUMENT;
			r_error.argument = 1;
			r_error.expected = from.get_type();
			return Variant();
		}
```